src/main/java/io/jenkins/plugins/coverage/source/DefaultSourceFileResolver.java

```diff
@@ -105,9 +115,28 @@ private String sanitizeFilename(String inputName) {
         return inputName.replaceAll("[^a-zA-Z0-9-_.]", "_");
     }
 
+    private Map<String, FilePath> createSourceFileMapping(FilePath workspace, TaskListener listener) {
+        try {
+            return Arrays
+                    .stream(workspace.list("**/*.java"))
```

Comment:
Ahh sorry, I totally mean to change that from `**/*.java` to `**/*`, but i'm happy to make the changes you've suggested if you think that's preferable? Just to verify what it is you're looking for there: in `tryFindSourceFile`, instead of looking at the wildcard every time, we check to find the file in our list of possible paths. If we can't find it there, we load it with the wildcard and add the parent path to the list of possible paths? 

Does doing that also resolve your other comment about doing the calculating in the agent instead of master? I guess it does?

Reply:
> Just to verify what it is you're looking for there: in tryFindSourceFile, instead of looking at the wildcard every time, we check to find the file in our list of possible paths. If we can't find it there, we load it with the wildcard and add the parent path to the list of possible paths?

Yes.
> Does doing that also resolve your other comment about doing the calculating in the agent instead of master? I guess it does?
Yes :)

> It feels like that wouldn't give us anywhere near the performance benefits that the above change does (assuming I change **/*.java to **/*. What are your thoughts on that?

Yeah, you are right, that will not increase the performance compare with your changes. But, in your implementation, you used the file name as the key of `sourceFileMapping`. And `sourceFilePath` might not be a file name instead of a path of the source file(like Cobertura). You won't get a correct source file form the map in this situation.

Reply:
@cizezsy Ok thanks. I'll take a look and see if there's any way I can fix it for Jacoco and keep the performance benefits without effecting the other implementations :)

Reply:
@cizezsy This change should have no impact on Corbertura at all. Previously at the end of `tryFindSourceFile`, we were just returning null. That will be exactly the same if the source file mapping isn't found in our map. Also, `tryFindSourceFile` has lots of code to handle the case where the source file is absolute, so Corbertura shouldn't need the fallback anyway I don't think. We have about 7000 classes, and this change has improved performance from about 10 minutes to 5 seconds. We'd really like it if we could keep those performance benefits :crossed_fingers:

Reply:
@bgreen1005 Sounds reasonable to me :p 

I will test the changes on my computer. If there are no other problems, we can get this merged. Thanks~

Reply:
@cizezsy Perfect! Thank you :slightly_smiling_face: Hopefully no issues :crossed_fingers:

Reply:
@cizezsy How's it looking? Have you had chances to check this? :slightly_smiling_face:

Reply:
@bgreen1005 I will try it this weekend. Sorry for the delay, I am very busy this week :(

Reply:
@cizezsy no problem :slightly_smiling_face: Thanks for the heads up.